### PR TITLE
Fix incorrect message size validation when 'REJECT' status is set

### PR DESCRIPTION
**Pull request by yuliasherman**
When non-primary replicas send a PreProcessReply message with the status "REJECT" to the primary, it does not contain a pre-execution result hash. So, the validate() function should expect to receive a short-sized message.

### CODE REVIEW

**Comment by teoparvanov** [comment]
Hi @HristoStaykov could you please take a look?

File: bftengine/src/preprocessor/messages/PreProcessReplyMsg.cpp [hunk 1]
```diff
@@ -42,19 +42,25 @@ void PreProcessReplyMsg::validate(const ReplicasInfo& repInfo) const {
...
-    throw runtime_error(__PRETTY_FUNCTION__ + string(": verifySig"));
-}
+  if (msgHeader.status == STATUS_GOOD) {
+    if (size() < (sizeof(Header) + sigLen)) {
+      LOG_ERROR(logger(),
+                "Message size is too small" << KVLOG(
+                    msgHeader.senderId, msgHeader.clientId, msgHeader.reqSeqNum, size(), sizeof(Header) + sigLen));
+      throw runtime_error(__PRETTY_FUNCTION__ + string(": Message size is too small"));
+    }
+    concord::diagnostics::TimeRecorder scoped_timer(*preProcessorHistograms_->validateMessage);
```
Comment (teoparvanov) [inline bftengine/src/preprocessor/messages/PreProcessReplyMsg.cpp:55]: shouldn't this be inside the if ?

**Review by teoparvanov — APPROVED** [review]
LGTM!

File: bftengine/src/preprocessor/messages/PreProcessReplyMsg.cpp (hunk 1 above)
Comment (yuliasherman) [reply bftengine/src/preprocessor/messages/PreProcessReplyMsg.cpp:55]: It should not be inside "if" as it measures how long verifySig takes

**Pull request merged**